### PR TITLE
Run shellcheck on scripts

### PR DESCRIPTION
**Pull request by szepeviktor**
## Description
Check common problems in our shell code.
@ramsey Should I go on and fix reported problems?

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR checklist
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by ramsey** [comment]
> Should I go on and fix reported problems?

Sounds good to me. Please do.

Be careful where it mentions needing to double quote to prevent globbing, etc. There are a few places where I'm intentionally not double-quoting, but if you're able to find a way that makes shellcheck happy and keeps the tests passing, then awesome!

**Comment by szepeviktor** [comment]
> There are a few places where I'm intentionally not double-quoting

It is very easy

```
"${var[@]}" === "arg1" "arg2" "arg3"
"${var[*]}" === "arg1 arg2 arg3"
```

**Comment by szepeviktor** [comment]
Please also see  #215

**Comment by szepeviktor** [comment]
It is harder than I thought! 🥊 

Not quoting is used to split string at whitespaces.

**Comment by szepeviktor** [comment]
How about making a key this way?

```bash
function make_key {
    tr --squeeze-repeats '\t ' '-' <<<"${*}"
}
```

**Comment by szepeviktor** [comment]
The above 👆 function does not exclude empty string elements of an array 😿

**Comment by szepeviktor** [comment]
... this feel like my private war against unquoted variables 🪖

**Comment by ramsey** [comment]
> How about making a key this way?

I was trying to go for a pure Bash solution, so I don't have to rely on external tools.

**Comment by ramsey** [comment]
The shell scripts need to run on GitHub's Windows environments, so any tools that don't exist there aren't going to work.

File: bin/composer_install.sh [hunk 1]
```diff
@@ -19,16 +19,12 @@ case "${dependency_versions}" in
     *) composer_command="install" ;;
 esac
 
-IFS=' '
-additional_options=("${additional_composer_options}")
-for option in "${additional_options[@]}"; do
-    composer_options+=("${option}")
-done
+read -r -a additional_options <<<"${additional_composer_options}"
```
Comment (ramsey) [inline bin/composer_install.sh:22]: Oh, nice! I didn't think about using redirection with `read`. I was thinking of it only in terms of user interaction.

**Review by ramsey — APPROVED** [review]
This is great! Thanks!

**Comment by szepeviktor** [comment]
Thank you!

While working on it I've realized Bash should handle command line arguments through `getopts`.

**Pull request merged**